src/AutoMapper/Mapper.cs

```diff
@@ -155,8 +161,18 @@ public Mapper(IConfigurationProvider configuration, Factory serviceCtor)
     ResolutionContext IInternalRuntimeMapper.DefaultContext => DefaultContext;
     Factory IInternalRuntimeMapper.ServiceCtor => _serviceCtor;
     public IConfigurationProvider ConfigurationProvider => _configuration;
-    public TDestination Map<TDestination>(object source) => Map(source, default(TDestination));
-    public TDestination Map<TDestination>(object source, Action<IMappingOperationOptions<object, TDestination>> opts) => Map(source, default, opts);
+    public TDestination Map<TDestination>(object source)
+    {
+        ArgumentNullException.ThrowIfNull(source);
```

Comment:
Really? :)

Reply:
Ha yes. We should be checking for nulls here when we rely on a not-null source object for getting the type.